common/src/main/java/org/apache/celeborn/common/network/protocol/TransportMessage.java

```diff
@@ -44,4 +57,38 @@ public int getMessageTypeValue() {
   public byte[] getPayload() {
     return payload;
   }
+
+  public <T extends GeneratedMessageV3> T getPayLoad() throws InvalidProtocolBufferException {
+    switch (messageTypeValue) {
+      case OPEN_STREAM_VALUE:
+        return (T) PbOpenStream.parseFrom(payload);
+      case STREAM_HANDLER_VALUE:
+        return (T) PbStreamHandler.parseFrom(payload);
+      default:
+        logger.error("Unexpected type {}", type);
+    }
+    return null;
+  }
+
+  public ByteBuffer toByteBuffer() {
+    int totalBufferSize = payload.length + 4 + 4;
+    ByteBuffer buffer = ByteBuffer.allocate(totalBufferSize);
+    buffer.putInt(messageTypeValue);
+    buffer.putInt(payload.length);
+    buffer.put(payload);
+    buffer.flip();
+    return buffer;
+  }
+
+  public static TransportMessage fromByteBuffer(ByteBuffer buffer) throws CelebornIOException {
+    int type = buffer.getInt();
+    if (MessageType.forNumber(type) == null) {
+      throw new CelebornIOException("Decode failed,fallback to legacy messages.");
```

Comment:
```suggestion
      throw new CelebornIOException("Decode failed, fallback to legacy messages.");
```

Reply:
Yes, it should be. I'll check other strings.